crates/iceberg/src/spec/snapshot.rs

```diff
@@ -124,6 +150,69 @@ impl Snapshot {
         Utc.timestamp_millis_opt(self.timestamp_ms).unwrap()
     }
 
+    /// Get the schema id of this snapshot.
+    #[inline]
+    pub fn schema_id(&self) -> Option<SchemaId> {
+        self.schema_id
+    }
+
+    /// Get the schema of this snapshot.
+    pub fn schema(&self, table_metadata: &TableMetadata) -> Result<SchemaRef> {
+        Ok(match self.schema_id() {
+            Some(schema_id) => table_metadata
+                .schema_by_id(schema_id)
+                .ok_or_else(|| {
+                    Error::new(
+                        ErrorKind::DataInvalid,
+                        format!("Schema with id {} not found", schema_id),
+                    )
+                })?
+                .clone(),
+            None => table_metadata.current_schema().clone(),
+        })
+    }
+
+    /// Get parent snapshot.
+    pub fn parent_snapshot(&self, table_metadata: &TableMetadata) -> Option<SnapshotRef> {
```

Comment:
Do we need to expose this API right now? It seems like it's only used in the test so what we could do is just use the table metadata to lookup the parent_snapshot_id which is already exposed.

Reply:
Fixed.